### PR TITLE
🚸 Ensure classId is in collection before viewing

### PR DESCRIPTION
**Pull request by AuroraHuang22**

### CODE REVIEW

File: src/pages/nft/claim/index.vue [hunk 1]
```diff
@@ -1223,14 +1230,47 @@ export default {
...
+        });
+
+        collectedNFTs = this.getCollectedNFTClassesByAddress(
+          this.claimingAddress
+        );
+
+        attempts += 1;
+
+        // eslint-disable-next-line no-await-in-loop
+        await new Promise(resolve => setTimeout(resolve, 500));
```
Comment (nwingt) [inline src/pages/nft/claim/index.vue:1258]: We have a sleep util
https://github.com/likecoin/liker-land/blob/4d007cd7731600803f9da21371209ef14fffff32/src/util/misc.js#L1-L3

File: src/pages/nft/claim/index.vue [hunk 2]
```diff
@@ -1223,14 +1230,47 @@ export default {
...
+
+      while (
+        !collectedNFTs?.some(item => item.classId === this.classId) &&
+        attempts < MAX_ATTEMPT
+      ) {
+        // eslint-disable-next-line no-await-in-loop
+        await this.fetchCollectedNFTClassesByAddress({
+          address: this.claimingAddress,
+          nocache: true,
+        });
```
Comment (nwingt) [inline src/pages/nft/claim/index.vue:1249]: Should try catch this?

File: src/pages/nft/claim/index.vue [hunk 3]
```diff
@@ -1223,14 +1231,49 @@ export default {
...
+
+          collectedNFTs = this.getCollectedNFTClassesByAddress(
+            this.claimingAddress
+          );
+
+          if (collectedNFTs?.some(item => item.classId === this.classId)) {
+            break;
+          }
+          // eslint-disable-next-line no-await-in-loop
+          await sleep(500);
```
Comment (williamchong) [inline src/pages/nft/claim/index.vue:1257]: The fetch above with `nocache: true` set a qs according to current utm second, so better sleep >= 1000ms

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/pages/nft/claim/index.vue (hunk 3 above)
Comment (AuroraHuang22) [reply src/pages/nft/claim/index.vue]: [🎨 Increase sleep duration](https://github.com/likecoin/liker-land/pull/1773/commits/e49acdb1a431c7fe1b8337f7c7e4730662d7cfdf)

File: src/pages/nft/claim/index.vue [hunk 4]
```diff
@@ -490,8 +490,12 @@
           <ButtonV2
             :content-class="['px-[48px]']"
             class="phoneLg:w-full phoneLg:max-w-[480px]"
-            preset="tertiary"
-            :text="$t('nft_claim_claimed_button_view_collection')"
+            :preset="isViewCollectionLoading ? 'plain' : 'tertiary'"
+            :text="
+              isViewCollectionLoading
+                ? $t('settings_follow_loading')
```
Comment (nwingt) [inline src/pages/nft/claim/index.vue:496]: Can we not reuse another translation key for this?

Comment (AuroraHuang22) [reply src/pages/nft/claim/index.vue]: [🎨 Update translation key](https://github.com/likecoin/liker-land/pull/1773/commits/ab945b9d23ad54c29e105caabb85adc149dea51b)

**Review by williamchong — APPROVED** [review]

**Pull request merged**